app/src/test/java/com/onboarding/calculatorkotlin/CalculatorTest.kt

```diff
@@ -0,0 +1,265 @@
+package com.onboarding.calculatorkotlin
+
+import com.nhaarman.mockitokotlin2.mock
+import com.onboarding.calculatorkotlin.mvp.CalculatorContract
+import com.onboarding.calculatorkotlin.mvp.model.CalculatorModel
+import com.onboarding.calculatorkotlin.mvp.presenter.CalculatorPresenter
+import com.onboarding.calculatorkotlin.util.ConstantsUtils.Result
+import junit.framework.TestCase.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.verifyNoMoreInteractions
+
+class CalculatorTest {
+    private lateinit var model: CalculatorContract.Model
+    private lateinit var presenter: CalculatorContract.Presenter
+    private var view: CalculatorContract.View = mock()
+
+    @Before
+    fun setup() {
+        model = CalculatorModel()
+        presenter = CalculatorPresenter(model, view)
+    }
+
+    @Test
+    fun setACorrectOperator() {
+        presenter.onOperatorButtonPressed(ADD)
+        assertEquals(ADD, model.getLastModified())
+        verify(view).showCompleteOperation(model.getOperation())
+        verify(view).showLastValue(model.getLastModified())
+        verifyNoMoreInteractions(view)
+    }
+
+    @Test
+    fun setValuesInFirstOperand() {
+        presenter.onNumberButtonPressed(ONE_STRING)
+        assertEquals(ONE_STRING, model.getLastModified())
+        verify(view).showCompleteOperation(model.getOperation())
+        verify(view).showLastValue(model.getLastModified())
+        verifyNoMoreInteractions(view)
+    }
+
+    @Test
+    fun setValuesInSecondOperand() {
+        model.setValue(ONE_STRING)
+        model.setOperator(ADD)
+        presenter.onNumberButtonPressed(TWO_STRING)
+        assertEquals(TWO_STRING, model.getLastModified())
+        verify(view).showLastValue(model.getLastModified())
+        verify(view).showCompleteOperation(model.getOperation())
+        verifyNoMoreInteractions(view)
+    }
+
+    @Test
+    fun CleanAllTheOperation() {
+        presenter.onCleanButtonLongPressed()
+        assertEquals(EMPTY_STRING, model.getOperation())
+        verify(view).resetResultView()
+        verify(view).showDeleteAllMessage()
+        verify(view).resetOperationView()
+        verifyNoMoreInteractions(view)
+    }
+
+    @Test
+    fun CleanAValueInSecondOperand() {
+        model.setValue(ONE_STRING)
+        model.setOperator(MUL)
+        model.setValue(TWO_STRING)
+        model.setValue(THREE_STRING)
+        presenter.onCleanButtonPressed()
+        verify(view).showCompleteOperation(model.getOperation())
+        verify(view).showLastValue(model.getLastModified())
+        verify(view).showDeleteMessage()
+        assertEquals(OPERATION_CLEAN_A_VALUE_IN_SECOND_OPERAND, model.getOperation())
+        verifyNoMoreInteractions(view)
+    }
+
+    @Test
+    fun CleanAValueInFirstOperand() {
```

Comment:
PLs change to lowerCase all the Test methods

Reply:
Solved!